packages/react-kit/src/components/Popover/Popover.tsx

```diff
@@ -145,70 +164,77 @@ class RawPopover extends React.Component<TFullPopoverProps, TPopoverState> {
 	}
 
 	render() {
-		const { closeOnClickAway, theme, hasArrow, onMouseDown, isOpened, onRequestClose, anchor } = this.props;
-
-		if (!isOpened || !anchor) {
+		const {
+			anchor,
+			closeOnClickAway,
+			duration = 0,
+			isOpened,
+			hasArrow,
+			theme,
+			onMouseDown,
+			onRequestClose,
+		} = this.props;
+
+		if (!anchor) {
 			return null;
 		}
 
 		const { top, left, arrowOffset, finalPlacement, finalAlign } = this.state;
-
-		const isMeasured = !!finalAlign && !!finalPlacement;
-
-		let style;
-		let popoverClassName = classnames(theme.container);
-		if (isMeasured && finalPlacement) {
-			style = prefix({
-				transform: `translate(${left || 0}px, ${top || 0}px)`,
-			});
-			popoverClassName = classnames(
-				popoverClassName,
-				{
-					[theme.container_hasArrow as string]: hasArrow,
-				},
-				[getPlacementModifier(finalPlacement)].map(mod => theme[`container_${mod}`]),
-			);
-		}
-		style = {
-			...style,
+		const isMeasured = finalAlign && finalPlacement;
+		const style = {
+			...(isMeasured &&
+				prefix({
+					transform: `translate(${left || 0}px, ${top || 0}px)`,
+				})),
 			...this.props.style,
 		};
-
-		let child = (
-			<BoundsUpdateDetector onUpdate={this.onSizeUpdate}>
-				<div
-					ref={(el: any) => (this._popover = el)}
-					style={style}
-					onMouseDown={onMouseDown}
-					onClick={stopPropagation}
-					className={popoverClassName}>
-					<div className={theme.content}>
-						{isMeasured &&
-							finalPlacement &&
-							finalAlign &&
-							hasArrow && (
-								<div
-									className={theme.arrow}
-									style={getArrowStyle(finalPlacement, finalAlign, arrowOffset)}
-								/>
-							)}
-						{this.props.children}
-					</div>
-				</div>
-			</BoundsUpdateDetector>
+		const popoverClassNames = classnames(
+			theme.container,
+			isMeasured && { [theme.container_hasArrow as string]: hasArrow },
+			isMeasured &&
+				finalPlacement &&
+				[getPlacementModifier(finalPlacement)].map(mod => theme[`container_${mod}`]),
 		);
-
-		if (closeOnClickAway) {
-			child = <RootClose onRootClose={onRequestClose}>{child}</RootClose>;
-		}
-
 		const target = typeof window !== 'undefined' ? window : 'window';
-
-		return (
+		const child = (state: string) => (
```

Comment:
It is render-helper. It useful only in method "render". If move it as instance method, this seem as it useful in other instance methods. This can lead to delusions.

Reply:
But you are recreating this function on each render. Most functions are used only in `render` is React components. That's not the reason for having them all inside `render` method.

Reply:
ok
fix it in 1e5e60de2a5292508290efa6ec28c3f474eedbe2